tilavarauspalvelu/api/graphql/types/application_round/serializers.py

```diff
@@ -36,3 +36,26 @@ def save(self, **kwargs: Any) -> ApplicationRound:
         instance = super().save(**kwargs)
         generate_reservation_series_from_allocations.delay(instance.pk)
         return instance
+
+
+class SetApplicationRoundResultsSentSerializer(NestingModelSerializer):
+    instance: ApplicationRound
+
+    class Meta:
+        model = ApplicationRound
+        fields = [
+            "pk",
+        ]
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        if self.instance.status != ApplicationRoundStatusChoice.HANDLED:
+            msg = "Application round is not in handled status."
+            raise ValidationError(msg, code=error_codes.APPLICATION_ROUND_NOT_HANDLED)
+
+        return attrs
+
+    def save(self, **kwargs: Any) -> ApplicationRound:
+        kwargs["sent_date"] = local_datetime()
+        instance = super().save(**kwargs)
```

Comment:
This should maybe be updated in `send_application_handled_emails`, so regardless of where the task is gets, it still gets updated (we already update `application.results_ready_notification_sent_date` there`)

Reply:
No, it needs to be here so that the status of the application round is updated when the mutation is called, so that frontend can change how the UI looks to match. The email sending is done in the background and takes time.